### ISSUE TITLE
Unable to remove uploaded presentation

### ISSUE THREAD

**Issue by bhuvankrishna** [bigbluebutton/bigbluebutton#12386]
**Describe the bug**
Unable to remove uploaded presentation. When clicking the trash icon nothing happens.

**To Reproduce**
1. Open Actions > manage presentations > Upload presentation
2. Now again Open Actions > manage presentations and click the trash icon
3. Nothing happens and the presentation still stays

**Expected behavior**
Once trash icon is selected the presentation should be removed

**BBB version (optional):**
BigBlueButton continually evolves. Providing the version/build helps us to pinpoint when an issue was introduced.
Example:
$ sudo bbb-conf --check | grep BigBlueButton
BigBlueButton Server 2.3.1 (2209)

**Desktop (please complete the following information):**
 - OS: Debian Unstable
 - Browser Firefox
 - Version 89

**Smartphone (please complete the following information):**
 - Device: Huawei P30 lite
 - OS: Android 10
 - Browser: firefox
 - Version: 88.1

**Additional context**

**Comment by hiroshisuga** [comment]
cannot confirm the phenomenon either on desktop or on smaltphone (although mine is P10lite). 
Please check if you can reproduce the issue on test23.bigbluebutton.org.

**Comment by bhuvankrishna** [comment]
On test23.bigbluebutton.org too I am unable to remove the default presentation. When I click the tick icon nothing happens. It is supposed to be not shown when I click the tick icon.
![Screenshot_20210525_191433](https://user-images.githubusercontent.com/592449/119508693-91dab500-bd8d-11eb-8a53-b6c162e26d1e.png)

**Comment by bhuvankrishna** [comment]
I think this issue is related to #12103 and #12237

**Comment by hiroshisuga** [comment]
It is not very clear to me what you want to do. So you can remove the uploaded presentation as the title reads? If so, you may want to close the issue..

**Comment by AlphaDE** [comment]
If it is the last (and only) presentation shown, in 2.3.1 you can't. 

The windows even shows the trashcan icon, but if you click it and the click confirm, nothing happens, the presentation stays open and in the list. If not needed anymore, the presentation window needs to be individually minimized by each and every participant.

**Comment by bhuvankrishna** [comment]
@AlphaDE Yes exactly you have captured the issue exactly.

**Comment by hiroshisuga** [comment]
OK, so the issue is to remove the default presentation (or to hide the whiteboard on each client), this is a duplicate of #12103 and #12237, as @bhuvankrishna has already pointed out.

**Comment by AlphaDE** [comment]
@hiroshisuga Not exactly... the issue is to remove the last presentation in the list (whether it is the default presentation or not) which does not work although a trash icon is shown. Deleting the last presentation should consequently lead to closing the whiteboard as there is nothing to show anymore.
To delete the last presentation in the list during a conference, another presentation needs to be uploaded and activated. This is not logical.

**Comment by hiroshisuga** [comment]
Hmm, Sorry but I cannot reproduce what you say in the test23 server... A movie will make your point clear. Re-submitting this as a new issue with different title will also help. Current title is a bit confusing.

**Comment by AlphaDE** [comment]
The test23 has a default presentation (which cannot be deleted as per different issue). On my server, there is no default presentation anymore, nevertheless, when I upload a single presentation, I cannot delete afterwards without uploading a second one.
test23 is so far not helpful for reproducing the issue.
Take a server, delete the default presentation in the file system so it comes up with no presentations, upload a üresentation and try to delete it. 
That's what I try to do and that's what's not working.

**Comment by hiroshisuga** [comment]
I know what you mean now.

I cannot help you a lot, but when I read the source code:
https://github.com/bigbluebutton/bigbluebutton/blob/3bfa5979e482348c5b7f4166c840b87b00fde222/bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx#L735
BBB seems to show the trash icon only when the file name is default.pdf (or the one that is specified in the configuration). And as far as I read the document (https://docs.bigbluebutton.org/2.2/customize.html#change-the-default-presentation), removing the default presentation is not really what the user is supposed to do. Only replacing it is allowed... 
Thus, in my view this issue is about removing the default presentation and a duplicate of #12103 and #12237.  

If you want to change the source code by yourself, 
https://github.com/bigbluebutton/bigbluebutton/blob/3bfa5979e482348c5b7f4166c840b87b00fde222/bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx#L393
seems to be where BBB bans zero presentation. You can change 0 to -1, but you will need to change many other places to avoid errors that may stop the server. I can only wish you a good luck...

**Comment by bhuvankrishna** [comment]
@hiroshisuga 

> The test23 has a default presentation (which cannot be deleted as per different issue). On my server, there is no default presentation anymore, nevertheless, when I upload a single presentation, I cannot delete afterwards without uploading a second one.
> test23 is so far not helpful for reproducing the issue.
> Take a server, delete the default presentation in the file system so it comes up with no presentations, upload a üresentation and try to delete it.
> That's what I try to do and that's what's not working.

I totally agree with you on this. Even on the self hosted server where we removed the default presentation. The presentation doesn't show when starting the room, But when we upload a presentation then we are stuck with it for the entire session since we can't remove it and continue the session without presentation. This was not the case with 2.2.x of BBB where we can remove all the presentations.

**Comment by bhuvankrishna** [comment]
> I know what you mean now.
> 
> I cannot help you a lot, but when I read the source code:
> 
> https://github.com/bigbluebutton/bigbluebutton/blob/3bfa5979e482348c5b7f4166c840b87b00fde222/bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx#L735
> 
> BBB seems to show the trash icon only when the file name is default.pdf (or the one that is specified in the configuration). So if you upload a file that is no named default.pdf, a trash icon is anyway shown, regardless of its functionality.
> 
> And as far as I read the document (https://docs.bigbluebutton.org/2.2/customize.html#change-the-default-presentation), removing the default presentation is not really what the user is supposed to do. Only replacing it is allowed... In fact I was surprised that BBB works without a presentation.
> 
> Thus, in my view this issue is about removing the default presentation and a duplicate of #12103 and #12237.
> 
> If you want to change the source code by yourself,
> 
> https://github.com/bigbluebutton/bigbluebutton/blob/3bfa5979e482348c5b7f4166c840b87b00fde222/bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx#L393
> 
> seems to be where BBB bans zero presentation. You can change 0 to -1, but you will need to change many other places to avoid errors that may stop the server. I can only wish you a good luck...

Thanks for the pointer @hiroshisuga. I will try it out hope it will solve the issue.

**Comment by bhuvankrishna** [comment]
> > I know what you mean now.
> > I cannot help you a lot, but when I read the source code:
> > https://github.com/bigbluebutton/bigbluebutton/blob/3bfa5979e482348c5b7f4166c840b87b00fde222/bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx#L735
> > 
> > BBB seems to show the trash icon only when the file name is default.pdf (or the one that is specified in the configuration). So if you upload a file that is no named default.pdf, a trash icon is anyway shown, regardless of its functionality.
> > And as far as I read the document (https://docs.bigbluebutton.org/2.2/customize.html#change-the-default-presentation), removing the default presentation is not really what the user is supposed to do. Only replacing it is allowed... In fact I was surprised that BBB works without a presentation.
> > Thus, in my view this issue is about removing the default presentation and a duplicate of #12103 and #12237.
> > If you want to change the source code by yourself,
> > https://github.com/bigbluebutton/bigbluebutton/blob/3bfa5979e482348c5b7f4166c840b87b00fde222/bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx#L393
> > 
> > seems to be where BBB bans zero presentation. You can change 0 to -1, but you will need to change many other places to avoid errors that may stop the server. I can only wish you a good luck...
> 
> Thanks for the pointer @hiroshisuga. I will try it out hope it will solve the issue.

One more thing in 2.2.x I was able to remove the presentation. So is it that in 2.3.x this feature has been removed?

**Comment by bhuvankrishna** [comment]
> Hey @bhuvankrishna ,do you have a good solution by now?
> We also need bbb most of the time without presentation. We have no default.pdf and can not delete last of uploaded new presentaions.
> 
> Is a change in this behavour planned?
> I dont get the point why there have to be one.

I have not tried the solution yet. I will keep posted once I try it out.

**Comment by sonerd** [comment]
Can confirm this issue. So removing a presentation and disabling the whiteboard with that is not possible in 2.3.x
